tool/tctl/common/edit_command.go

```diff
@@ -0,0 +1,178 @@
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// EditCommand implements the `tctl edit` command for modifying
+// Teleport resources.
+type EditCommand struct {
+	app    *kingpin.Application
+	cmd    *kingpin.CmdClause
+	config *service.Config
+	ref    services.Ref
+}
+
+func (e *EditCommand) Initialize(app *kingpin.Application, config *service.Config) {
+	e.app = app
+	e.config = config
+	e.cmd = app.Command("edit", "Edit a Teleport resource")
+	e.cmd.Arg("resource type/resource name", `Resource to update
+	<resource type>  Type of a resource [for example: rc]
+	<resource name>  Resource name to update
+	
+	Example:
+	$ tctl edit rc/remote`).SetValue(&e.ref)
+}
+
+func (e *EditCommand) TryRun(ctx context.Context, cmd string, client auth.ClientI) (match bool, err error) {
+	if cmd != e.cmd.FullCommand() {
+		return false, nil
+	}
+
+	f, err := os.CreateTemp("", "teleport-resource*.yaml")
+	if err != nil {
+		return true, trace.Wrap(err)
+	}
+
+	defer func() {
+		if err := os.Remove(f.Name()); err != nil {
+			fmt.Fprintf(os.Stderr, "ERROR: could not remove temporary file %v\n", f.Name())
+		}
+	}()
+
+	rc := &ResourceCommand{
+		refs:     services.Refs{e.ref},
+		format:   teleport.YAML,
+		stdout:   f,
+		filename: f.Name(),
+		force:    true,
+	}
+	rc.Initialize(e.app, e.config)
+
+	err = rc.Get(ctx, client)
+	if closeErr := f.Close(); closeErr != nil {
+		return true, trace.Wrap(err)
+	}
+	if err != nil {
+		return true, trace.Wrap(err, "could not get resource %v: %v", rc.ref.String(), err)
+	}
+
+	originalSum, err := checksum(f.Name())
+	if err != nil {
+		return true, trace.Wrap(err)
+	}
+
+	args := strings.Fields(editor())
+	editorCmd := exec.CommandContext(ctx, args[0], append(args[1:], f.Name())...)
+	editorCmd.Stdin = os.Stdin
+	editorCmd.Stdout = os.Stdout
+	editorCmd.Stderr = os.Stderr
+	if err := editorCmd.Start(); err != nil {
+		return true, trace.BadParameter("could not start editor %v: %v", editor(), err)
+	}
+	if err := editorCmd.Wait(); err != nil {
+		return true, trace.BadParameter("skipping resource update, editor did not complete successfully: %v", err)
+	}
+
+	newSum, err := checksum(f.Name())
+	if err != nil {
+		return true, trace.Wrap(err)
+	}
+
+	// nothing to do if the resource was not modified
+	if changed := newSum != originalSum; !changed {
+		return true, nil
+	}
+
+	newName, err := resourceName(f.Name())
+	if err != nil {
+		return true, trace.Wrap(err)
+	}
+
+	if e.ref.Name != newName {
+		return true, trace.NotImplemented("renaming resources is not supported with tctl edit")
+	}
+
+	if err := rc.Create(ctx, client); err != nil {
+		return true, trace.Wrap(err)
+	}
+
+	return true, nil
+}
+
+// editor gets the text editor to be used for editing the resource
+func editor() string {
+	for _, v := range []string{"TELEPORT_EDITOR", "VISUAL", "EDITOR"} {
```

Comment:
Nit: Move `TELEPORT_EDITOR` into a constant e.g. like we do [in tsh](https://github.com/gravitational/teleport/blob/master/tool/tsh/tsh.go#L479-L484).

Reply:
I kind of like that all 3 literals are right here, so you don't have to go searching around for them and you can see the order of precedence. Since `TELEPORT_EDITOR` isn't used anywhere else I don't see much value in the constant.